python/lsst/drp/tasks/gbdesAstrometricFit.py

```diff
@@ -57,6 +58,66 @@
 ]
 
 
+def calculate_apparent_motion(data, refEpoch):
+    """Calculate shift from reference epoch to the apparent observed position
+    at another date.
+
+    This function calculates the shift due to proper motion combined with the
+    apparent motion due to parallax. This is not used in the
+    `GbdesAstrometricFitTask` or related child tasks, but is useful for
+    assessing results.
+
+    Parameters
+    ----------
+    data : `pd.DataFrame`
+        Table containing position, proper motion, parallax, and epoch for each
+        source.
+    refEpoch : `float`
+        Epoch of the reference position.
+
+    Returns
+    -------
+    apparentMotionRA : `np.ndarray` [`astropy.units.Quantity`]
+        RA shift in degrees.
+    apparentMotionDec : `np.ndarray` [`astropy.units.Quantity`]
+        Dec shift in degrees.
+    """
+    ra_rad = (data["ra"].values * u.deg).to(u.rad)
+    dec_rad = (data["dec"].values * u.deg).to(u.rad)
+
+    dt = (astropy.time.Time(data["MJD"].values, format="mjd") - astropy.time.Time(refEpoch, format="mjd")).to(
+        u.yr
+    )
+    pmRA_deg = (data["pmRA"].values * u.mas).to(u.deg) / u.yr
+    pmDec_deg = (data["pmDec"].values * u.mas).to(u.deg) / u.yr
+    properMotionRA = pmRA_deg * dt
+    properMotionDec = pmDec_deg * dt
+
+    obsTimes = astropy.time.Time(data["MJD"], format="mjd")
```

Comment:
Why not assing this above, and then use it in `dt`? Otherwise, you're making this time object twice. Or does it make more sense to have the observed epoch be a separate input argument, as a Time object already?

Reply:
I think I just missed that I was using the identical conversion twice.